### ISSUE TITLE
ValidationError contains an infinite recursion

### ISSUE THREAD

**Issue by diezztsk** [Automattic/mongoose#5208]
**Bug**

This issue is similar to [#1834](https://github.com/Automattic/mongoose/issues/1834) and [#1296](https://github.com/Automattic/mongoose/issues/1296)

I have NodeJs v7.8.0 and mongoose v4.9.6
Code to reproduce an error is:
```js
const mongoose = require('mongoose');
const Schema   = mongoose.Schema;

const AdditionalContactSchema = new Schema({
    contactName: {
        type: String,
        required: true
    },
    contactValue: {
        type: String,
        required: true
    }
});

const ContactSchema = new Schema({
    name: {
        type: String,
        required: true
    },
    email: {
        type: String,
        required: true
    },
    additionalContacts: [AdditionalContactSchema]
});

const EmergencyContactSchema = new Schema({
    contactName: {
        type: String,
        required: true
    },
    contact: ContactSchema
});

EmergencyContact = mongoose.model('EmergencyContact', EmergencyContactSchema);

(async() => {
    let contact = new EmergencyContact({
        contactName: 'Electrical Service',
        contact: {
            name: 'John Smith',
            email: 'john@gmail.com',
            additionalContacts: [
                {
                    contactName: 'skype',
                    // Forgotten value
                }
            ]
        }
    });

    try {
        await contact.validate();
    } catch (err) {
        // In this place there is an error with infinite depth.
        console.log(JSON.stringify(err));
    }
})();
```
ValidationError in this case have infinite structure. And if I try to convert into json I receive an error: `TypeError: Converting circular structure to JSON`

If I try to use model with one nested level then ValidationError have't infinite depth. For example:
```js
const mongoose = require('mongoose');
const Schema   = mongoose.Schema;

const AdditionalContactSchema = new Schema({
    contactName: {
        type: String,
        required: true
    },
    contactValue: {
        type: String,
        required: true
    }
});

const EmergencyContactSchema = new Schema({
    contactName: {
        type: String,
        required: true
    },
    name: {
        type: String,
        required: true
    },
    email: {
        type: String,
        required: true
    },
    additionalContacts: [AdditionalContactSchema]
});

EmergencyContact = mongoose.model('EmergencyContact', EmergencyContactSchema);

(async() => {
    let contact = new EmergencyContact({
        contactName: 'Electrical Service',
        name: 'John Smith',
        email: 'john@gmail.com',
        additionalContacts: [
            {
                contactName: 'skype',
                // Forgotten value
            }
        ]
    });

    try {
        await contact.validate();
    } catch (err) {
        // All work fine.
        console.log(JSON.stringify(err));
    }
})();
```

**Comment by SBRK** [comment]
I can confirm I a similar issue, with v4.9.3

**Comment by vkarpov15** [comment]
Thanks for the detailed repro script, will be fixed with 4.9.8.
